File: chap1/sec1.5/prob1.1/sol_stab.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Wed Sep 12 17:08:31 2018

Script to compute the eigenvalue for the Jacobian of the Lorenz System
"""
import numpy as np

# Set Lorenz system parameters
sigma, rho, beta = 10, 28, 8/3

K = (beta*(rho-1))**0.5

# Build the Jacobian matrix at one fixed point
mJac = np.array([[-sigma, sigma, 0],
                 [1, -1, -K],
                 [K, K, -beta]])

# Compute the eigenvalues
vLam = np.linalg.eigvals(mJac)
nLam = len(vLam)

# Print the eigenvalues
for i in range(nLam):
    lam = vLam[i]
    print('lam{} = {:1.2f} + {:1.2f}j'.format(i, np.real(lam), np.imag(lam)))

File: chap1/sec1.5/prob1.1/sol_stab.m
% Set Lorenz system parameters
sigma = 10; rho = 28; beta = 8/3;

K = (beta*(rho-1))^0.5;

% Build the Jacobian matrix at one fixed point
mJac = [[-sigma, sigma, 0];
        [1, -1, -K];
        [K, K, -beta]];

% Compute the eigenvalues
vLam = eig(mJac);
nLam = length(vLam);

% Print the eigenvalues
for i = 1:nLam
    lam = vLam(i);
    fprintf('lam%d = %1.2f + %1.2fj\n', i, real(lam), imag(lam));
end